tests/unit/algorithms/action/adapters/openvino/test_action_dataloader.py

```diff
@@ -0,0 +1,384 @@
+"""Unit Test for otx.algorithms.action.adapters.openvino.dataloader."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from typing import List, Optional
+
+import numpy as np
+import pytest
+
+from otx.algorithms.action.adapters.openvino.dataloader import (
+    ActionOVClsDataLoader,
+    ActionOVDemoDataLoader,
+    ActionOVDetDataLoader,
+    _is_multi_video,
+    get_ovdataloader,
+)
+from otx.api.entities.annotation import (
+    Annotation,
+    AnnotationSceneEntity,
+    AnnotationSceneKind,
+)
+from otx.api.entities.dataset_item import DatasetItemEntity
+from otx.api.entities.datasets import DatasetEntity
+from otx.api.entities.id import ID
+from otx.api.entities.image import Image
+from otx.api.entities.label import Domain, LabelEntity
+from otx.api.entities.metadata import MetadataItemEntity, VideoMetadata
+from otx.api.entities.scored_label import ScoredLabel
+from otx.api.entities.shapes.rectangle import Rectangle
+from tests.test_suite.e2e_test_system import e2e_pytest_unit
+
+
+class MockDatasetEntity(DatasetEntity):
+    """Mock class for DatasetEntity."""
+
+    def __init__(self):
+        pass
+
+
+class MockDatasetItemEntity(DatasetItemEntity):
+    """Mock class for DatasetItemEntity"""
+
+    def __init__(
+        self,
+        metadata: List[MetadataItemEntity],
+        annotation_scene: Optional[AnnotationSceneEntity] = None,
+    ):
+        self.__metadata = metadata
+        self.__annotation_scene: AnnotationSceneEntity = annotation_scene
+
+    def get_metadata(self) -> List[MetadataItemEntity]:
+        return self.__metadata
+
+
+class MockImage(Image):
```

Comment:
This is a just question. Why don't you use `MockImage` class in tests/unit/algorithms/action/test_helpers.py but instead re-define it?

Reply:
I didn't expect previous PR is merged before I create this PR.
Thanks for updating me. I change codes to remove some duplicated implementations